pkg/instrumentation/sdk.go

```diff
@@ -171,6 +171,18 @@ func (i *sdkInjector) injectCommonSDKConfig(ctx context.Context, otelinst v1alph
 			Value: chooseServiceName(pod, resourceMap, appIndex),
 		})
 	}
+
+	idx = getIndexOfEnv(container.Env, constants.EnvOTELServiceVersion)
```

Comment:
Can we put this behind a featuregate? ([ex](https://github.com/open-telemetry/opentelemetry-operator/blob/87d4e103b2484fbedb7232d124c6e5cd7b2c84b7/pkg/instrumentation/podmutator.go#L89))

Reply:
Thank you for the feedback, I'll work this into the change :)

Reply:
This is a great feature. Will it be enabled by default?

Reply:
Hey, thanks!  I'd be happy to leave this on by default.  Any objections?

Reply:
None from me! :bow: